### DISCUSSION TITLE
PDF file does not open

### DISCUSSION THREAD

**Discussion by EsraaSalahAli** [QuestPDF/QuestPDF#661] [Q&A]
Hello,
I am trying to use the quest pdf package to generate pdf file from web api project.
the pdf file result from the code does not open and show me that message "Adobe Acrobat Reader could not open 'hello.pdf' because it is either not a supported file type or because the file has been damaged (for example, it was sent as an email attachment and wasn't correctly decoded)."
so i tried to debug the code and it stops at this point => byte[] pdfBytes = document.GeneratePdf(); every time without throwing any errors or exception and does not complete the following code after this point.
I don't know what is the problem and here is my code snippet

        var document = Document.Create(container =>
        {
        
            container.Page(page =>
            {
        
               page.Size(PageSizes.A4);
        
                page.Header().Text("Invoice").FontSize(30).Bold();
        
                page.Content().Text($"Invoice Number: {invoice.InvoiceNum}");
                page.Content().Text($"Customer Name: {invoice.Client.NameAr}");
                page.Content().Text($"Date: {invoice.Date}");
                page.Content().Text($"Total: {invoice.Total}");
       
                page.Content().Table(table =>
                {
                    table.ColumnsDefinition(columns =>
                    {
                        columns.RelativeColumn();
                        columns.RelativeColumn();
                        columns.RelativeColumn();
                    });
        
            table.Header(header =>
                    {
                        header.Cell().Text("Description");
                        header.Cell().Text("Quantity");
                        header.Cell().Text("Price");
                    });

                    foreach (var item in invoice.InvoiceItems)
                    {
                        table.Cell().Text(item.ProductDescription);
                        table.Cell().Text(item.Quantity.ToString());
                        table.Cell().Text(item.UnitPrice.ToString());
                    }
                });

                page.Footer().Text("Thank you for your business!");
            });
        });
        byte[] pdfBytes = document.GeneratePdf();
        MemoryStream ms = new MemoryStream(pdfBytes);

**Comment by MarcinZiabek** [comment]
Please make sure that your byte stream is appropriately and fully written to the designated file 😀 Most likely it is buffered and not written when the application closes.

**Comment by EsraaSalahAli** [comment]
The problem solved when i used older version of the package

**Comment by DawidRobertNowak** [comment]
Hello,

Glad you were able to find the solution! We are closing this thread as resolved. Thank you for sharing your answer — it may help others with the same question.

Thank you!
